src/bid/encoding.rs

```diff
@@ -123,7 +127,7 @@ pub(crate) fn preimage_gadget(
     let type_fields = BlsScalar::from_bytes(&TYPE_FIELDS).unwrap();
 
     // Add to the composer the values required for the preimage.
-    let mut messages: Vec<Variable> = vec![];
+    let mut messages: Vec<Variable> = Vec::new();
     messages.push(composer.add_input(type_fields));
     // Push cipher as scalars.
     messages.push(encrypted_data.0);
```

Comment:
Not sure why you remove the macro, the `alloc` crates also have it, we used in the past. Plus, in this specific case, you're creating a new vector and *then* you're immediately push elements into it: this is not performant, the `vec!` macro is created exactly for such job in mind. So it would be more performant have:
```rust
let messages: Vec<Variable> = vec![
    composer.add_input(type_fields)),
    encrypted_data.0,
    encrypted_data.1,
    // etc
];
```